jobRun.go

```diff
@@ -0,0 +1,284 @@
+package jobsd
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/simpleframeworks/logc"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+// JobRun .
+type JobRun struct {
+	ID                    int64 `gorm:"primaryKey"`
+	OriginID              int64 `gorm:"index"`
+	Name                  sql.NullString
+	NameActive            sql.NullString `gorm:"unique"`
+	Job                   string
+	JobArgs               Args
+	Delay                 time.Duration
+	RunAt                 time.Time
+	RunCount              int
+	RunLimit              sql.NullInt64
+	RunStartedAt          sql.NullTime
+	RunStartedBy          sql.NullInt64 `gorm:"index"`
+	RunCompletedAt        sql.NullTime
+	RunCompletedError     sql.NullString
+	RetryTimeout          time.Duration
+	RetryTimeoutAt        sql.NullTime `gorm:"index"`
+	RetriesOnErrorCount   int
+	RetriesOnErrorLimit   int
+	RetriesOnTimeoutCount int
+	RetriesOnTimeoutLimit int
+	Schedule              sql.NullString
+	ClosedAt              sql.NullTime
+	ClosedBy              sql.NullInt64 `gorm:"index"`
+	CreatedAt             time.Time     `gorm:"index"`
+	CreatedBy             int64
+}
+
```

Comment:
I did this deliberately because I wanted to separate the concerns. This struct is primarily representative of the actual DB row. It also represents an instance of a Job i.e a JobRun. Whereas the JobContainer is merely a template for creating and updating a template for JobRuns.

Retries are their own beast. I think you made a comment about that below (i'll explain there)

Reply:
Fair enough!